specs/select.spec.md

```diff
@@ -0,0 +1,156 @@
+# Select + Select-Option
+
+## Overview
+
+The `select` is a component that provides a list of options for the user to select from.
+
+### Background
+
+There is a standard select html control, `<select>` and `<option>`, but it doesn't support the full level of detail, styling, etc that most design systems need or require. For context please see [MDN Docs](https://developer.mozilla.org/en-US/docs/web/html/element/select).
+
+### Use Cases
+
+- *A customer using the component on a web page.*
+On a web page a customer is shopping for a new shirt in size medium. The customer opens the size `select` and sees a list of `<option>`s and selects size medium.
+
+### Features
+- form association
+- focus delegation
+
+### Features
+- Select one or multiple `<fast-option/>`(s) using the `multiple` attribute
+- When Dropdown does not have enough screen real estate to open below `<fast-select>` it will open above.
+
+### Prior Art/Examples
+- [FAST Select (React)](https://www.npmjs.com/package/@microsoft/fast-components-react-msft)
+- [Ant Design](https://ant.design/components/select/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/select/code/)
+- [Atlassian UI](https://atlaskit.atlassian.com/packages/core/select)
+- [Lightning Design System](https://www.lightningdesignsystem.com/components/select/)
+- [W3 Example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html)
+
+---
+
+### API
+
+*Component Name*:
+- `fast-select`
+
+*Attributes:*
+- `autocomplete` - Allows the developer to provide a hint on how to search the content within the `<fast-option>`(s)
+- `autofocus` - automatically focuses the control
+- `disabled` - disables the control
+- `length` - The number of `<fast-option>` elements in the `<fast-select>` element
+- `multiple` - Allows user to select more than one `<fast-option>`
+- `name` - the name of the control
+- `options` - Returns a `HTMLOptionsCollection` of the `<fast-option>` elements contained by the `<fast-select>` element
+- `required` - boolean value that sets the field as required
+- `selectedIndex` - The index of the first or last selected `<fast-option>` element, depending on the value of `multiple`
+- `selectedOptions` - An `HTMLCollection` of the selected `<fast-option>` elements
+- `size` - If the `<fast-select>` is shown with a scrollbar, this represents how many `<fast-option>`s are visible.
+- `value` - string value of the number field, can be an empty string
```

Comment:
Would be nice if this wasn't forced to a string.

Reply:
this is currently how `value` works (in the platform) and raises an interesting question with regards to Open UI conformance. If you intend to have value take <n> types rather than following `name=foo[]` it's worth discussing how to handle this divergence.

Reply:
It's something that I've seen as broken with form controls for a long time. In real apps, we often have to abstract over value just to maintain the type, for example. So, if I've got a list of Person objects and I want to put them in a select, I want the selected value to be the person instance. Folks will often get around this by using the id of an object but then they have to write extra code in their models to do lookups and convert back and forth. Beyond that, not every model has an id of course.

Reply:
For text input, I expect value to be string. For a number input, I expect a number. For a select or listbox, I expect the value to be any object type. For a date picker, I expect a date object, etc.